tern/analyze/common.py

```diff
@@ -452,3 +453,61 @@ def get_os_style(image_layer, binary):
                     package_manager=binary,
                     package_format=image_layer.pkg_format,
                     os_list=image_layer.os_guess), 'info'))
+
+
+def check_git_src(dockerfile_path, src_path):
```

Comment:
Seems to me you don't need `src_path` here. You can get the path to the git repository from the directory where the Dockerfile is.

Reply:
Do we still need to parse `ADD` command to get the `src_path`? since we don't need `src_path` in this function.
I am thinking if `find_git_info` in tern/analyze/docker/dockerfile.py is necessary, because it seems that we don't need to know what is being copied by `ADD` command, and what we use is the `dockerfile path` which is given by a `dfobj`.

Reply:
So it looks to me like `find_git_info` does the ADD command parsing, and you want to check if what got added is a git repository in itself. In this case, you will need functionality to:
1. See if src_path exists (if os.path.exists(src_path))
2. If it does, see if it is a git repository (get_git_toplevel(src_path))
This is only useful if you can produce a full repository url for the git project, otherwise you just get whatever was in the ADD command.
So I think in this case, you do not have to worry about parsing the ADD command to get the `src_path`. We just want to see if the Dockerfile lives in a github repo.

Reply:
Maybe i can keep the current `find_git_info` which implements parsing ADD command, and not use the parsed `src_path`?

Reply:
Yup! In fact, you don't need to parse it out right now. You can work on that part for #585.